### PR TITLE
Add support for Atziri's Rule

### PR DESCRIPTION
**Pull request by Nostrademous**
Implement Atziri's Rule functionality.

Additionally Adds:
- Support for Unleash for triggers that can be supported by it (applies to all triggers, not just Queen's Demand)
- Config Option to set `UseMaxSeals` for skills that have Unleash on the source trigger

### CODE REVIEW

**Comment by Ydoum** [comment]
Any update on this?

**Comment by Nostrademous** [comment]
Fixes #2677

File: src/Modules/ModParser.lua [hunk 1]
```diff
@@ -1462,6 +1462,15 @@ local function triggerExtraSkill(name, level, noSupports)
...
 end
+local function triggerExtraSkillWithSource(name, level, sourceSkill, noSupports)
+	name = name:gsub(" skill","")
+	sourceSkill = sourceSkill:gsub(" skill","")
+	if gemIdLookup[name] then
+		return {
+			mod("ExtraSkill", "LIST", { skillId = gemIdLookup[name], level = level, noSupports = noSupports, triggered = true, source = sourceSkill })
+		}
+	end
+end
```
Comment (Wires77) [inline src/Modules/ModParser.lua:1473]: Was there a reason this couldn't be combined with `triggerExtraSkill`?

File: src/Modules/CalcPerform.lua [hunk 2]
```diff
@@ -2506,13 +2508,26 @@ function calcs.perform(env, avoidCache)
...
+			else
+				ConPrintf("[ERROR]: Unhandled Unique Trigger Name: " .. uniqueTriggerName)
 			end
 		end
 		if not source or #spellCount < 1 then
-			env.player.mainSkill.skillData.triggeredByUnique = nil
-			env.player.mainSkill.infoMessage = s_format("No %s Triggering Skill Found", triggerName)
-			env.player.mainSkill.infoMessage2 = "DPS reported assuming Self-Cast"
-			env.player.mainSkill.infoTrigger = ""
+			if not specialCase then
```
Comment (Wires77) [inline src/Modules/CalcPerform.lua:2525]: ?

File: src/Modules/CalcPerform.lua [hunk 3]
```diff
@@ -2506,13 +2508,26 @@ function calcs.perform(env, avoidCache)
...
 				end
+			elseif uniqueTriggerName == "Queen's Demand" then
+				triggerName = env.player.mainSkill.activeEffect.grantedEffect.name
+				if skill.activeEffect.grantedEffect.name == uniqueTriggerName then
+					source, trigRate = findTriggerSkill(env, skill, source, trigRate)
+				end
+				if skill.skillData.triggeredByUnique and env.player.mainSkill.socketGroup.slot == skill.socketGroup.slot then
+					t_insert(spellCount, { uuid = cacheSkillUUID(skill), cd = cooldownOverride or (skill.skillData.cooldown / icdr), next_trig = 0, count = 0 })
+				end
+				env.player.mainSkill.infoTrigger = ""
```
Comment (Wires77) [inline src/Modules/CalcPerform.lua:2519]: I don't think this is needed
```suggestion
				env.player.mainSkill.infoTrigger = ""
```

File: src/Modules/CalcOffence.lua [hunk 4]
```diff
@@ -1453,6 +1453,24 @@ function calcs.offence(env, actor, activeSkill)
...
+				for _, triggerSkill in ipairs(actor.activeSkillList) do
+					if cacheSkillUUID(triggerSkill) == skillData.triggerSourceUUID then
+						calcs.buildActiveSkill(env, calcMode, triggerSkill)
+						break
+					end
+				end
+				local cachedSourceSkill = GlobalCache.cachedData[calcMode][skillData.triggerSourceUUID]
+				-- if properly processed, get it's dpsMultiplier to increase triggerRate
+				if cachedSourceSkill then
+					skillData.triggerRate = skillData.triggerRate * (cachedSourceSkill.ActiveSkill.skillData.dpsMultiplier or 1)
```
Comment (Wires77) [inline src/Modules/CalcOffence.lua:1469]: Can we add this to the breakdown, too?  Eff. Trigger Rate on the Calcs tab doesn't end up showing one, but it's higher than the others and people will probably want to know where it comes from.

Didn't realize triggered skills could be supported by unleash, actually

**Review by Wires77 — CHANGES_REQUESTED** [review]
Looks pretty good, needs a couple of cleanup commits is all

File: src/Modules/CalcPerform.lua (hunk 2 above)
Comment (Nostrademous) [reply src/Modules/CalcPerform.lua:2525]: leftover accident... always true ... just removing

File: src/Modules/CalcPerform.lua (hunk 3 above)
Comment (Nostrademous) [reply src/Modules/CalcPerform.lua:2519]: correct, another left over

**Pull request closed without merging**